docs/specs/authentication-flow.md

```diff
@@ -356,6 +356,58 @@ reauth_flows:
     - authentication: secondary_sms_code
 ```
 
+### Flow selection
+
+By default, authentication flow with name `default` is used. Multiple flows per type can be specified in the configuration.
+
+Example:
+
+```yaml
+authentication_flow:
+  signup_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  login_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+  signup_login_flows:
+    - name: oauth_only
+      steps:
+      # ...
+    - name: email_password_2fa
+      steps:
+      # ...
+
+oauth:
+  clients:
+    - client_id: public_app
+      # Only allow OAuth login for public_app
+      x_authentication_flows_allowlist:
+      - type: login
+        name: oauth_only
+      - type: signup_login
+        name: oauth_only
+    - client_id: internal
+      # Allow both OAuth login and email_password_2fa for internal
+      x_authentication_flows_allowlist:
```

Comment:
Not sure if one flow is less secure than the other, should we need to allow triggering a stricter flow on important actions e.g. delete account